File: 2928-distribute-candies-among-children-i/2928-distribute-candies-among-children-i.rb
# @param {Integer} n
# @param {Integer} limit
# @return {Integer}
def distribute_candies(n, limit)
  (0..limit).sum do |i|
    min_candies = [0, n - i - limit].max
    max_candies = [limit, n - i].min
    (max_candies >= min_candies) ? (max_candies - min_candies + 1) : 0;
  end
end

File: 2928-distribute-candies-among-children-i/2928-distribute-candies-among-children-i.cpp
class Solution {
public:
    int distributeCandies(int n, int limit) {
        int res=0;
        for (int i=0; i <= limit; ++i) {
            int minCandies = max(0, n - i - limit);
            int maxCandies = min(limit, n - i);
            res += (maxCandies >= minCandies) ? (maxCandies - minCandies + 1) : 0;
        }
        return res;
    }
};